pyonfx/timestamps.py

```diff
@@ -0,0 +1,552 @@
+
+        Inspired by: https://gitlab.com/mbunkus/mkvtoolnix/-/blob/72dfe260effcbd0e7d7cf6998c12bb35308c004f/src/merge/timestamp_factory.cpp#L201-267
+
+        Parameters:
+            file_content (TextIOWrapper): The timestamps content
+            version (int): The version of the timestamps (only 2 or 4 is allowed)
+            rounding_method (RoundingMethod): A rounding method
+
+        Returns:
+            A tuple containing these 3 informations:
+                1. A list of each timestamps rounded to milliseconds
+                2. The last timestamps not rounded
+                3. The fpms
+        """
+
+        if version != 2 and version != 4:
+            raise ValueError("You can only specify version 2 or 4.")
+
+        timestamps: List[int] = []
+        previous_timestamp: int = 0
+        lowest_timestamp: Fraction = None
+        highest_timestamp: Fraction = None
+
+        for line in file_content:
+            line = line.strip(" \t")
+
+            if not line or line.startswith("#"):
+                continue
+
+            try:
+                timestamp = Fraction(line)
+            except ValueError:
+                raise ValueError(
+                    f'The timestamps file contain a invalid line. Here is it: "{line}"'
+                )
+
+            if highest_timestamp is None or highest_timestamp < timestamp:
+                highest_timestamp = timestamp
+            if lowest_timestamp is None or lowest_timestamp > timestamp:
+                lowest_timestamp = timestamp
+
+            rounded_timestamp = rounding_method(timestamp)
+
+            if version == 2 and rounded_timestamp < previous_timestamp:
+                raise ValueError(
+                    f"The timestamps file contain timestamps NOT in ascending order."
+                )
+
+            previous_timestamp = rounded_timestamp
+            timestamps.append(rounded_timestamp)
+
+        if not len(timestamps):
+            raise ValueError(f"The timestamps file is empty.")
+
+        if version == 4:
+            timestamps.sort()
+
+        duration = highest_timestamp - lowest_timestamp
+        if duration:
+            fpms = Fraction(len(timestamps) - 1) / (
+                highest_timestamp - lowest_timestamp
+            )
+        else:
+            fpms = 0
+        return timestamps, highest_timestamp, fpms
+
+
+class Timestamps:
+    """Timestamps object contains informations about the timestamps of an video.
+
+    Video player have 2 methods to deal with timestamps. Some floor them and other round them.
+    This can lead to difference when displaying the subtitle.
+        Ex:
+            Player - Method - proof
+            mpv    - round  - https://github.com/mpv-player/mpv/blob/7480efa62c0a2a1779b4fdaa804a6512aa488400/sub/sd_ass.c#L499
+            FFmpeg - floor  - https://github.com/FFmpeg/FFmpeg/blob/fd1712b6fb8b7acc04ccaa7c02b9a5c9f233cfb3/libavfilter/vf_subtitles.c#L195
```

Comment:
I guess that you are talking about ffmpeg.
Like you can see [here](https://github.com/libass/libass/blob/9f4e6afeec54c0d240a9b8377f27e6c5b7c2bde0/libass/ass.h#L605-L606), the type of ``now`` is ``long long``, but ffmpeg call it with a ``double``. This means it will be truncated which is equivalent to floor.

Reply:
Gotcha
Then it is probably more appropriate to put:
https://github.com/FFmpeg/FFmpeg/blob/fd1712b6fb8b7acc04ccaa7c02b9a5c9f233cfb3/libavfilter/vf_subtitles.c#L194-L196

Reply:
I updated the doc with your link